src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -85,10 +107,11 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
-  
+
   /** Returns YouTube video ID for a given {@code videoUrl}. */
   private String getVideoId(String videoUrl) {
-    String pattern = "(?<=watch\\?v=|/videos/|embed\\/|youtu.be\\/|\\/v\\/|\\/e\\/|watch\\?v%3D|watch\\?feature=player_embedded&v=|%2Fvideos%2F|embed%\u200C\u200B2F|youtu.be%2F|%2Fv%2F)[^#\\&\\?\\n]*";
+    String pattern =
+        "(?<=watch\\?v=|/videos/|embed\\/|youtu.be\\/|\\/v\\/|\\/e\\/|watch\\?v%3D|watch\\?feature=player_embedded&v=|%2Fvideos%2F|embed%\u200C\u200B2F|youtu.be%2F|%2Fv%2F)[^#\\&\\?\\n]*";
     Pattern compiledPattern = Pattern.compile(pattern);
```

Comment:
You could also make the Pattern a field that's set in `init` using the regex constant so it isn't recompiled every request.

Reply:
fixed